packages/backend-auth/src/factory.ts

```diff
@@ -126,12 +126,24 @@ class AmplifyAuthGenerator implements ConstructContainerEntryGenerator {
 
     const authConstructMixin: BackendAuth = {
       ...authConstruct,
+      /**
+       * Returns a resourceAccessAcceptor for the given role
+       * @param roleName Either the auth or unauth role name or the name of a UserPool group
+       */
       getResourceAccessAcceptor: (
-        roleName: AuthRoleName
+        roleName: AuthRoleName | string
```

Comment:
It's no longer a roleName if it can be a UserPool group name as well

Reply:
I couldn't really think of a better name for the input. Maybe `roleIdentifier`?

Reply:
It's not an exposed API right? If it isn't how about just 
`getResourceAccessAcceptor: (identifier: AuthRoleName | string) => ResourceAccessAcceptor`

Reply:
It's exposed in plugin-types. I didn't see your comment until after I changed it to `roleIdentifier`. If you feel strongly, I can change the name to just `identifier`.
Worth noting that the prop name can be changed at anytime because it doesn't affect the actual method signature.